crates/chain-config/src/config/codec/decoder.rs

```diff
@@ -0,0 +1,170 @@
+use std::{
+    fmt::Debug,
+    fs::File,
+    io::Read,
+    path::{
+        Path,
+        PathBuf,
+    },
+};
+
+use super::parquet;
+use itertools::Itertools;
+
+use crate::{
+    config::{
+        contract_balance::ContractBalance,
+        contract_state::ContractStateConfig,
+    },
+    CoinConfig,
+    ContractConfig,
+    Group,
+    MessageConfig,
+    StateConfig,
+};
+
+use super::GroupResult;
+
+pub enum IntoIter<T> {
+    InMemory {
+        groups: std::vec::IntoIter<GroupResult<T>>,
+    },
+    Parquet {
+        decoder: parquet::Decoder<File, T>,
+    },
+}
+
+impl<T> Iterator for IntoIter<T>
+where
+    parquet::Decoder<File, T>: Iterator<Item = GroupResult<T>>,
+{
+    type Item = GroupResult<T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self {
+            IntoIter::InMemory { groups } => groups.next(),
+            IntoIter::Parquet { decoder } => decoder.next(),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+enum DataSource {
+    Parquet {
+        snapshot_dir: PathBuf,
+    },
+    InMemory {
+        state: StateConfig,
+        group_size: usize,
+    },
+}
+
+#[derive(Clone, Debug)]
+pub struct Decoder {
+    data_source: DataSource,
+}
+
+impl Decoder {
+    pub fn json(
+        snapshot_dir: impl AsRef<Path>,
+        group_size: usize,
+    ) -> anyhow::Result<Self> {
+        let path = snapshot_dir.as_ref().join("state.json");
+
+        // This is a workaround until the Deserialize implementation is fixed to not require a
```

Comment:
It is fixed as part of the [v0.43.1](https://github.com/FuelLabs/fuel-vm/releases/tag/v0.43.1)